packages/math-poolsimulator/src/gyro3/index.test.ts

```diff
@@ -11,57 +11,74 @@ import { ExtendedGyro3 } from "./index";
 describe("Tests new Gyro 3CLP math function based on package other functions", () => {
   const poolData = poolsFromFile["pool"];
   const percentages = [25, 50, 75, 95];
+  const tokens = ["USDC", "BUSD", "USDT"];
 
-  const tokenIn = "USDC";
-  const tokenOut = "BUSD";
+  tokens.forEach((tokenIn) => {
+    tokens.forEach((tokenOut) => {
+      if (tokenIn === tokenOut) return; // Skip when both tokens are the same
 
-  percentages.forEach((percentage) => {
-    describe(`Tests for ${percentage}% with ${tokenIn} as tokenIn`, () => {
-      const amm = new AMM(
-        new ExtendedGyro3({
-          swapFee: poolData.swapFee,
-          totalShares: poolData.totalShares,
-          tokens: poolData.tokens,
-          tokensList: poolData.tokensList,
-          root3Alpha: poolData.root3Alpha,
-        }),
-      );
+      percentages.forEach((percentage) => {
+        describe(`Tests for ${percentage}% with ${tokenIn} as tokenIn and ${tokenOut} as tokenOut`, () => {
+          const amm = new AMM(
```

Comment:
is this diff right?

Reply:
yes I just moved the 

```
percentages.forEach((percentage) => {
    describe(`Tests for ${percentage}% with ${tokenIn} as tokenIn`, () => {
      const amm = new AMM(
        new ExtendedGyro3({
          swapFee: poolData.swapFee,
          totalShares: poolData.totalShares,
          tokens: poolData.tokens,
          tokensList: poolData.tokensList,
          root3Alpha: poolData.root3Alpha,
        }),
      );
```

inside and git made it weird kkkk